subsys/usb/device_next/class/usbd_uvc.c

```diff
@@ -542,22 +543,41 @@ static int uvc_get_vs_probe_frame_interval(const struct device *dev, struct uvc_
 {
 	struct uvc_data *data = dev->data;
 	struct uvc_format_descriptor *format_desc;
-	struct uvc_frame_discrete_descriptor *frame_desc;
-	int max;
+	struct uvc_frame_descriptor *frame_desc;
+	int min, max, max_id;
 
 	uvc_get_vs_fmtfrm_desc(dev, &format_desc, &frame_desc);
 	if (format_desc == NULL || frame_desc == NULL) {
 		LOG_DBG("Selected format ID or frame ID not found");
 		return -EINVAL;
 	}
 
+	if (frame_desc->bDescriptorSubtype == UVC_VS_FRAME_UNCOMPRESSED ||
+	    frame_desc->bDescriptorSubtype == UVC_VS_FRAME_MJPEG) {
+		struct uvc_frame_discrete_descriptor *desc =
+			(struct uvc_frame_discrete_descriptor *)frame_desc;
```

Comment:
To try to save one review cycle, I think there was a proposal to use `(void *)`:
```suggestion
		struct uvc_frame_discrete_descriptor *desc = (void *)frame_desc;
```

Reply:
Ok, fixing it.